echo-telemetry/src/main/java/com/netflix/spinnaker/echo/config/TelemetryConfig.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2019 Armory, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.echo.config;
+
+import com.netflix.spinnaker.echo.telemetry.TelemetryService;
+import com.netflix.spinnaker.retrofit.RetrofitConfigurationProperties;
+import com.netflix.spinnaker.retrofit.Slf4jRetrofitLogger;
+import com.squareup.okhttp.OkHttpClient;
+import java.util.concurrent.TimeUnit;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import retrofit.RestAdapter;
+import retrofit.client.OkClient;
+import retrofit.converter.JacksonConverter;
+
+@Slf4j
+@Configuration
+@ConditionalOnProperty("telemetry.enabled")
+@EnableConfigurationProperties(TelemetryConfig.TelemetryConfigProps.class)
+public class TelemetryConfig {
+
+  @Bean
+  public TelemetryService telemetryService(
+      RetrofitConfigurationProperties retrofitConfigurationProperties,
+      TelemetryConfigProps configProps) {
+    log.info("Telemetry service loaded");
+
+    TelemetryService client =
+        new RestAdapter.Builder()
+            .setEndpoint(configProps.endpoint)
+            .setConverter(new JacksonConverter())
+            .setClient(telemetryOkClient(configProps))
+            .setLogLevel(retrofitConfigurationProperties.getLogLevel())
+            .setLog(new Slf4jRetrofitLogger(TelemetryService.class))
+            .build()
+            .create(TelemetryService.class);
+
+    return client;
+  }
+
+  private OkClient telemetryOkClient(TelemetryConfigProps configProps) {
+    OkHttpClient httpClient = new OkHttpClient();
+    httpClient.setConnectTimeout(configProps.connectionTimeoutMillis, TimeUnit.MILLISECONDS);
```

Comment:
I think the timeouts here will help, and may be sufficient to have reasonable resilience against the endpoint going down.  (If we're only trying to log completed pipelines, those may be infrequent enough that the timeout is enough to save us from building up a huge backlog of outgoing connections.)  But I'd still suggest using a fault tolerance library, such as `Hystrix` (which we use extensively in gate) to circuit-break if this starts failing a lot---at the very least before we increase the scope of events that are logged.

Reply:
Ack - https://github.com/spinnaker/spinnaker/issues/4864